---
title: "Intro to Linux & R"
subtitle: ""
toc: false
date: ""
sidebar: false
image: "assets/images/seo.webp"
format:
  html:
    css: "./styles.css"
    number-sections: false
    title-block-banner: false
    page-layout: full
execute:
  freeze: false
---

Welcome to the Day1


## Intended Learning Outcomes

By the end of this session, participants will be able to:

*Unix/Linux*

1. **Install and verify** the functionality of key **genomics tools**: `Plink`, `vcftools`, `BCFtools`, and `Bedtools`.
1. **Navigate the Unix/Linux environment**, including creating directories, moving between locations, and managing files.
Extra:
1. Inspect and interpret VCF files, including understanding key fields and genotype formats.
1. Use vcftools and bcftools to filter, summarize, and manipulate VCF data.

*R* 

1. **Install and load essential R packages** for genomic analysis: `GAPIT3`, `ggplot2`, and `rrBLUP`.
1. Prepare input files for downstream analysis in R, including GAPIT-compatible formats.
1. **Load and explore data in R**, including reading tables and data frames, and performing basic plotting.

## Learning Activities

To achieve the above outcomes, participants will:

*Unix/Linux:*

1. Follow guided installation steps for Plink, vcftools, BCFtools, and Bedtools on their systems (see pre-course materials).
1. Run basic Unix commands to create directories, navigate folders, and manage files (1st meeting).
1. Explore a small VCF file in class to understand its structure and metadata(1st meeting).
1. Use vcftools and bcftools to filter variants, calculate allele frequencies, and extract summary statistics (extra).
1. Practice using bedtools to intersect VCF data with genomic regions(extra).
1. Convert VCF to Plink format and run basic Plink commands to check data quality(extra).

*R:*

1. Install R packages and verify functionality by loading them in R(1st meeting).
1. Load processed data into R, inspect it using `str()`, `summary()`, and `head()`, and visualize it using `ggplot2`(1st meeting).
1. Prepare genotype and phenotype files for GAPIT and run a basic association analysis (demonstrated by instructor).


## 🧰 `Before Day 1`
Read the `pre-course` ⬆️ and be prepared accordingly. You need following `tools` in the day 1:

::: {.callout-important collapse=true}
## 👉Tools

- `WSL`/`Unix`/`Ubuntu`/`Shell` [The Unix Shell](https://learn.microsoft.com/en-us/windows/wsl/install) for **windows users**.
-  `Plink`, `vcftools`, `BCFtools`, and `Bedtools` (see linux chapter).
- `R` (version ≥ 4.0)
- `RStudio`
- Recommended packages in RStudio: `GAPIT3`, `ggplot2`, and `rrBLUP`(see linux and R chapter).
- `Plink` at Linux (https://zzz.bwh.harvard.edu/plink/dist/plink-latest.zip)
:::

## 📅 `Schedule of Day 1`
| Day 1, Time                      | Topic                        | Instructor              |
|---------------------------|------------------------------------|----------------|
| 10.30-11.30| Unix (CLI) | Abu|
| 11.45-12.30| R & R Studio | Abu|
| 13.30-15.30| Lab (practical)| Abu| 



## 🟢`At Day 1` 
Linux, R, Lab. 
See up, bar headers ⬆️

## 📚 Course materieals
::: {.callout-important collapse=true}
### 👉️ [Course materieals]()

##### 1. Download this repository:

```bash
wget https://github.com/abu85/intro-to-linux-and-R/archive/refs/heads/main.zip . # download with 'wget' command
```

##### 2. Extract and enter into the repository folder📁
```bash

unzip main.zip            # extract everything from the zip file to folder

ls                        # see the file folder

cd intro-to-linux-and-R   # change and enter into the folder(directory)
```

Or Clone this repository if you have (git):
```bash
git clone https://github.com/YOUR_USERNAME/intro-to-R-genomic-selection.git
cd intro-to-linux-and-R
```

#### 📚 Contents of the 📁 **intro-to-linux-and-R**

```bash
ls -la         # see the content through linux
```
It will be look like as follows

`practicals`
`scripts`        

##### Important folders for you
- `slides/`: Lecture slides
- `scripts/`: Lecture R scripts for each topic
- [Practicals](./practicals/practical_01_intro.html): RMarkdown files for hands-on exercises
- `practicals/exercise/data/data_for_linux`: Sample genotype datasets
- `practicals/exercise/data/data_for_linux`: Sample phenotype datasets



::: {.callout-tip}
## Tip

## 📬 For questions or feedback, contact at `abu.siddique@slu.se`

:::


## License

This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](http://creativecommons.org/licenses/by-sa/4.0/).
