---
title: "Intro to Linux and R (practical)"
---

# Goals (to do list)

A) `Linux`: Practice basic Linux shell commands, load tools:

- open the terminal or wsl command line
- where am I (print working directory)
- list or view (different styles)
- create a directory (folder)
- change working directory to the new directory
- absolute/relative paths of that directory
- include options & arguments
- copy and paste files & folders
- move files & folders
- rename files & folders
- delete files or folders
- download tools (Plink, vcftoolf, bcftools etc)
- unzip or extract
- activate tool


B) `R`: Load packages, set working directories, Explore data in R:
- Setting a project or working directory
- Interacting with R
- Install Packages
- Assign values or objects
- Data types, structures
- How to import and export data or results
- Housekeeping

## A. `Linux`: 

### 1. Open Linux terminal at your PC

search `terminal` or `Ubuntu` or `command prompt` and click on the first result

### 2. List files
```bash
ls 
```

What else I can do with this command?
Options:
```bash
ls --help
```

```bash
man ls 
```
Note: press on **Q** keyboard button to **exit**

classify list (files or folders):
```bash
ls -F 
```

```bash
ls -F ./

```
Here `-F` is an *option* and `./` is an *argument* 
![](https://swcarpentry.github.io/shell-novice/fig/shell_command_syntax.svg)



list all:
```bash
ls -a
```

print the author of each file:
```bash
ls -la
```



### 3. Where am I?

What is my present working directory?

```bash
pwd
```


### 4. Create a folder/directory

Do it in home or current working directory

```bash
mkdir my_tools
```

multiple sub folders at once:
```bash
mkdir my_tools/plink
```
list them:
```bash
ls -a
```


### 5. Go in the new directory
```bash
cd my_tools
```
Or go to another sub directory
```bash
cd /plink
```


How do you know you moved in or not?

look at the texts before the `$` sign in the terminal

Or execute next command
```bash
pwd
```

Make a text file
```bash
touch test.txt

ls
```

copy and paste the file

```bash
cp test.txt test_copy.txt

ls
```

move file to a subfolder 
```bash
mv test.txt ./<subfolder>/test.txt
```

::: {.callout-tip title="TO DO example" collapse="false"}

What if you execute: 

`cd ..` and 

`cd -` and 

`cd --` 

What they do?

How to come back to the place(directory) where you were?

:::

### TO DO 1:
::: {.callout-tip title="TO DO 1" collapse="false"}

- create a *folder* and a *sub-folder*
- make a file (name: *file.txt*) in 1st folder
- move or copy it to the *sub-folder*
- print *content* of the sub-folder, see if *file.txt* is there
- rename the file to *file_v1.txt*
:::

### 6. Software instalation in Linux 

**must for this workshops**

::: {.callout-tip title="tools installation in Linux" collapse="false"}

### Download the official Plink softwere binary and make it available

```bash
cd --               # be in your home directory

cd my_tools/plink   # be in your plink directory
```
We use `wget` command for this 
see all version [here](https://www.cog-genomics.org/plink/)

```bash
wget https://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20250819.zip
```

#### Unzip the downloaded file
use `unzip` command, 
```bash
unzip plink_linux_x86_64.zip
```
to know more `unzip --help`

#### Make the binary executable: change the permision
use `chmod` command
```bash
chmod +x plink
```
to know more `chmod --help`


#### Add Plink to your PATH 
(*optional but recommended*)
```bash
echo 'export PATH=$PATH:~/my_tools/plink' >> ~/.bashrc
source ~/.bashrc
```
this command ensures you can run it from anywhere in the terminal just by typing plink, without needing to specify the full path.

Explanation: 
- `echo` = print what is after it

- `>>` = is a shell operator used to append text to a file.

- `~`  = *home/username*

- `.bashrc` is a configuration file in your home directory that runs every time you start a new terminal session (specifically for Bash shell).


#### Test installation
check version of the Plink tool
```bash
plink --version
```
You will see something printed as like as bellow: 

>PLINK v1.9.0-b.7.11 64-bit (19 Aug 2025)




If not that means Plink is not activated, we need activate it (repeat and fix *Make the binary executable: change the permision* step)


Or install this way:

```bash
sudo apt install putty-tools
```

Note: Some mac users may need to install java or update java


### Install vcftools, bcftools, bedtools
first see them if tools are installed already or not
```bash
vcftools --version
bcftools --version
bedtools --version
```
If you see *error* or there is no tools like that, then follow the rest of the commands and execute one by one. It will install tools.

```bash
sudo apt install vcftools  # install vcftools at your windows wsl or mac

vcftools --version
```

```bash
sudo apt install bcftools

bcftools --version
```

```bash
sudo apt install bedtools # install vcftools at your windows wsl or mac

bedtools --version
```

:::

Well done!




### TO DO 2: 
::: {.callout-tip title="TO DO 2" collapse="false"}

- download today's 'course materials' (it is a [repository in github](https://github.com/abu85/intro-to-linux-and-R/archive/refs/heads/main.zip))
- hints: use `wget` command
- extract the contents of the file (zip)
- change the working directory to that folder or directory
- solutions: see codes at 'Course materials' at ['Home page'](https://abu85.github.io/intro-to-linux-and-R/)
:::


### 7. create a file and write something in it


```bash
nano test.txt
```

copy and paste following line

```
>GCTAGTGCAATTT
>GTTAGTGCAATTT
```

`Ctrl` + `right click` mouse

>`Ctrl` + `o` for write out, then `Enter` to save the name, then `Ctrl` + `x` for exit




### TO DO 3

::: {.callout-tip title="TO DO 3" collapse="false"}

- edit the file `file.txt` and add the following lines in the file: 

```
>CCTAGGGCAATTT
>CCTAGGGCAATTT
>CGTAGTGCAATTT
>ACTAGGAACATTT
>GCTAGTGCAATTT
>ACTAGGAACATTT
>GCTAGTGCAATTT
```

- exit the file or exit from editing
- make it *executable*
:::

### 8. Word count (extra)
Print file's content
```bash
cat file.txt
```

count line of the words
```bash
wc -l file.txt
```


#### print and pipe

```bash
cat file.txt | wc -l
```


### TO DO 4 (extra)

::: {.callout-tip title="TO DO 4" collapse="false"}

- see the couple of lines (use `head`) 
- count number of lines in a `file.text` file
- use | pipe
- save output to a file called `result.txt` (use `>`)
:::

What you have learn so far:

::: {.callout-tip title="Linux commands" collapse="true"}

### Workshop oriented commands

`ls` ,`man`, `pwd`, `cd`, `mkdir`, `wget` (for plinks tool), `tar` or `unzip`, `nano`, `less`, `head`, `cp`, `mv`, `rm`, use of `\`, arguments.

### More commands

- `less` – view a file
- `nano` – view and edit a file
- `head` / `tail`
- `grep`
- `find`
- `wildcards`
- tab completion – use it!!!
- `man` – manual pages or
- anycommand `--help`
- `|`
:::


## B. `R`:

### 1. Packages needed for R modules

- `GAPIT3`
- `ggplot2`
- `rrBLUP`

::: {.callout-tip title="GAPIT3 installation" collapse="false"}
### GAPIT is a package that is run in the R software environment.
R can be freely downloaded from [http://www.r-project.org](http://www.r-project.org). 

We also recommend the integrated development environment RStudio which is also freely available at [http://www.rstudio.com](http://www.rstudio.com).


[GAPIT](https://github.com/jiabowang/GAPIT) can currently be installed in several ways.

- From source on the internet (remember you need to run it everytime you open RStudio)
- From GitHub
- From an archive


### Installation from source functions

GAPIT can be loaded with a single funciton. 


```r
source("http://zzlab.net/GAPIT/gapit_functions.txt")
```

Or from GitHub function.
```r
source("https://raw.githubusercontent.com/jiabowang/GAPIT/refs/heads/master/gapit_functions.txt", encoding = "UTF-8")
```

### Installation from GitHub

Installation can also be made from GitHub when the R package `devtools` is available.

```r
install.packages("devtools")
devtools::install_github("jiabowang/GAPIT", force=TRUE)
library(GAPIT)
```

or

```r
install.packages("remotes")
remotes::install_github("jiabowang/GAPIT")
library(GAPIT)
```

### Installation from an archive


GAPIT can be installed from an archive such as \*.tar.gz or \*.zip archive.
An archive can be downloaded from the "releases" page.
If you would like the latest version of GAPIT from the GitHub site you may want to clone it and then build it (this may require Rtools on Windows).

```bash
git clone git@github.com:jiabowang/GAPIT.git
R CMD build GAPIT
```

Once an archive has been obtained it can be installed from a shell, similar to as follows.


```bash
R CMD INSTALL GAPIT_3.5.0.9000.tar.gz
```

Or similarly from within R.

```
install.packages("GAPIT_3.5.0.9000.tar.gz", repos = NULL, type="source")
```

### In some case of the BiocManager can not be installed

Installation of same packages such as “multtest” and “biobase” can not be intalled from BiocManager directly. These packages can be downloaded in the Bioconductor website and be installed from local source files.

The website of Bioconductor is here:

```
https://bioconductor.org/packages/3.19/bioc/
```

More details please check and [ref user manual](https://github.com/jiabowang/GAPIT).

:::


### 2. Install **ggplot2** & **rrBLUP** packages
install packs
```r
install.packages("ggplot2")
install.packages("rrBLUP")
```
attach the packs
```r
library("ggplot2")
library("rrBLUP")
```


### 3. Setting a project or working directory

- to know (print) where you are right now, run this `getwd()` function at R Console 

- to set directory at your target directory, run this `setwd()` function at R Console with target path (see example bellow)

example:
```r
setwd("C:/Users/auue0001/OneDrive - Sveriges lantbruksuniversitet/path/to/your/folder")`
```
### TO DO 5

::: {.callout-tip title="TO DO 5" collapse="false"}

- print location, where you are  (current working directory)
- set working directories to the 'course materials' at your PC
:::

### 4. Interacting with R
Guided tour around the RStudio


### 5. How to import and export data or results

 - How to do that from RStudio? (guided tour)
 - How to do that with function or command?
 

Load a data from the course materials (manually and by code)
```r
library(readr)  # we need a library 

allele_freq <- read_delim(
"practicals/excercise/data/data_4_linux/allele_freq.frq", 
    delim = "\t", 
    escape_double = FALSE, 
    trim_ws = TRUE)

```
View the loaded dataset

```r
View(allele_freq)
```


Load data table from a website to R 
```{r}
# loading data set
myY=read.table(file="https://github.com/jiabowang/GAPIT/raw/refs/heads/master/Documents/mdp_traits.txt", head = TRUE)
```

### TO DO 6
::: {.callout-tip title="TO DO 6" collapse="false"}

- upload `example.vcf` data file in RStudio as 'vcf_data' object 
- have the column headers but not the comments. Hints: data has 'column headers' ( as #) and 'comments' (comments starts with `##`)
- view the data
:::

### 6. Data structures

Example R code to see the data structure:
```{r}
str(cars)
```
A data frame with 50 observations on 2 variables.

[,1]	speed	numeric	Speed (mph)
[,2]	dist	numeric	Stopping distance (ft)


Lets try `help` `?? cars`

Lets see structure of the allele_freq

```r
str(allele_freq)
```


### TO DO 7
::: {.callout-tip title="TO DO 7" collapse="false"}

- see & print the structure of the `vcf_data` object

:::
### 7. Housekeeping

Comment, Quarto , R history

### 8. Export data, results or plot

Export results or data or object

export as a csv file

```r
write.csv(df , file = "path/to/your/saving/folder/df.csv")
```

export as a excel file
```r
library(writexl)   # we need an extra library for exporting excel

writexl::write_xlsx (test_df, path = "C:/Users/your_username/test_df.xlsx", col_names = TRUE, format_headers = TRUE)
```



### 9. Plotting a figure (extra)
Speed and Stopping Distances of Cars
```r
p1 <- ggplot(data = cars, aes (x = speed, y = dist))+
geom_point()

p1
```

#### 9.1 Export plots

to export ggplot: 
```r 
ggsave("path/to/your/saving/plots/figure_1a.png", plot = p1, bg ="white")
```
to export base r plot: 

```r
dev.print (device=jpeg, file="path/to/your/saving/plots/figure_1a.jpg", width=par("din")[1]*300, res=300, quality=100)
```
### 10. R built-in function  

- Directory: `setwd`, `getwd` 

- Upload data to R: `read.table`, `read_table`, `read_csv` 

- See data structure: `str`, `summarize`, `summary`. 

- Session information: `sessionInfo()`. It summarizes all your used packs
--------------- 

Well done! Congrats!!

END